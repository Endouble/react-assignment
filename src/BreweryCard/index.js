import React, { Component } from 'react';
import PropTypes from 'prop-types';

import './index.scss';

class BreweryCard extends Component {
    constructor(props) {
        super(props);

        this.onClick = this.onClick.bind(this);
    }

    onClick(evt) {
        this.props.onClick(evt, this.props.brewery.id);
    }

    render() {
        const {
            brewery,
        } = this.props;

        return (
            <div
                tabIndex={0}
                className="Brewery-card"
                onFocus={() => {}}
                onKeyDown={() => {}}
                role="button"
                onClick={this.onClick}
            >
                {brewery.name}
            </div>
        );
    }
}

BreweryCard.defaultProps = {
    onClick: () => {},
};

BreweryCard.propTypes = {
    onClick: PropTypes.func,
    brewery: PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
    }).isRequired,
};

export default BreweryCard;
