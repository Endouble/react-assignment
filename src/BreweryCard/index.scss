.Brewery-card {
    background: #f7f7f7;
    padding: 20px;
    height: 100px;
    margin: 5px;
}
