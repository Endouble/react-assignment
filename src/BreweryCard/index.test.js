import React from 'react';
import renderer from 'react-test-renderer';
import BreweryCard from './index.js';

it('BreweryCard renders correctly', () => {
    const brewery = {
        "id": 2,
        "name": "Avondale Brewing Co",
        "brewery_type": "micro",
        "street": "201 41st St S",
        "city": "Birmingham",
        "state": "Alabama",
        "postal_code": "35222-1932",
        "country": "United States",
        "longitude": "-86.774322",
        "latitude": "33.524521",
        "phone": "2057775456",
        "website_url": "http://www.avondalebrewing.com",
        "updated_at": "2018-08-23T23:19:57.825Z"
    };

    const tree = renderer
        .create(<BreweryCard brewery={brewery} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
