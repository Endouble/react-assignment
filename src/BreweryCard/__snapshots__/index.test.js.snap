// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreweryCard renders correctly 1`] = `
<div
  className="Brewery-card"
  onClick={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  role="button"
  tabIndex={0}
>
  Avondale Brewing Co
</div>
`;
